### PR TITLE
Use any week standard

### LINKED ISSUE #67: Allow for multi-day/week/month intervals
Currently, incidence only allows multi-day intervals, but it's possible to allow multi-week or month intervals as well (and possibly hourly...?). This would require a change in the `get_interval()` method and the `valid_date_interval()` internal function.

### LINKED ISSUE #89: Add epiweek for interval option
Hey,

Speaking to people at WHO on the course and the standard epiweek from CDC starts on a sunday. It would be good if interval = "epiweek" could be added as one of the options for interval where it defaults to a week starting on a sunday. 

Possibly as well it would be nice to be able to put in Mon, Tues, Wed, Thurs ... to define the start day of the week. 

I know you can put the start_date to solve these problems but I think this would be easier for the end user.

### PR DESCRIPTION
**Pull request by zkamvar**
This update will fix #89 (which is the resurrection of #24) and fix #67 
 
Installation:

```r
remotes::install_github("reconhub/incidence#110")
```

 Here are the changes:

 - ISOweek import changed to [aweek](https://www.repidemicsconsortium.org/aweek)
 - Weekly intervals can start on any day of the week by allowing things like "epiweek", "isoweek", "wednesday week", "2 Saturday weeks", etc. (see https://github.com/reconhub/incidence/issues/55#issuecomment-405297526)
 - Any interval `seq.Date()` can handle (e.g. "5 weeks") can be handled by `incidence()` (#67)
 - isoweeks are still added to the incidence object, but are officially deprecated.
 - the item `$weeks` is now added to the incidence object, which contains an "aweek" class
 - plotting will now force the first tick to be the starting point of the incidence curve
 - tests have been added

Still needs to be done:

 - [x] examples and documentation in the functions about the week intervals
 - [x] vignette for handling weeks
 - [x] Update NEWS
 - [x] Bump version to 1.7.0
 - [ ] release to CRAN

### Example

``` r
library('incidence')
dat <- outbreaks::ebola_sim$linelist$date_of_onset
print(iso <- incidence(dat, "isoweek"))
#> <incidence object>
#> [5888 cases from days 2014-04-07 to 2015-04-27]
#> [5888 cases from ISO weeks 2014-W15 to 2015-W18]
#> 
#> $counts: matrix with 56 rows and 1 columns
#> $n: 5888 cases in total
#> $dates: 56 dates marking the left-side of bins
#> $interval: 1 week
#> $timespan: 386 days
#> $cumulative: FALSE
plot(iso)
```

``` r
plot(iso, labels_week = FALSE)
```

``` r

print(cdc <- incidence(dat, "MMWRweek"))
#> <incidence object>
#> [5888 cases from days 2014-04-06 to 2015-04-26]
#> [5888 cases from MMWR weeks 2014-W15 to 2015-W17]
#> 
#> $counts: matrix with 56 rows and 1 columns
#> $n: 5888 cases in total
#> $dates: 56 dates marking the left-side of bins
#> $interval: 1 week
#> $timespan: 386 days
#> $cumulative: FALSE
plot(cdc, labels_week = FALSE)
```

``` r
print(sat <- incidence(dat, "Saturday week"))
#> <incidence object>
#> [5888 cases from days 2014-04-05 to 2015-04-25]
#> [5888 cases from (Saturday) weeks 2014-W14 to 2015-W17]
#> 
#> $counts: matrix with 56 rows and 1 columns
#> $n: 5888 cases in total
#> $dates: 56 dates marking the left-side of bins
#> $interval: 1 week
#> $timespan: 386 days
#> $cumulative: FALSE
plot(sat, labels_week = FALSE)
```

``` r
print(sat2 <- incidence(dat, "2 Saturday weeks"))
#> <incidence object>
#> [5888 cases from days 2014-04-05 to 2015-04-18]
#> [5888 cases from (Saturday) weeks 2014-W14 to 2015-W16]
#> 
#> $counts: matrix with 28 rows and 1 columns
#> $n: 5888 cases in total
#> $dates: 28 dates marking the left-side of bins
#> $interval: 2 weeks
#> $timespan: 379 days
#> $cumulative: FALSE
plot(sat2, labels_week = FALSE)
```

<sup>Created on 2019-03-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### CODE REVIEW

**Comment by zkamvar** [comment]
It would also be nice if @aspina7 and @jakobschumacher could see if this works for them

**Comment by zkamvar** [comment]
If we want to get this onto CRAN relatively soon, this also needs to be merged before the end of the week so that we can submit it before the 20th.

File: R/plot.R [hunk 1]
```diff
@@ -110,28 +116,36 @@ plot.incidence <- function(x, ..., fit = NULL, stack = is.null(fit),
...
-        ylab <- "Biweekly incidence"
+        ylab <- "semi-weekly incidence"
       } else {
-        ylab <- sprintf("Incidence by period of %d days",
+        ylab <- sprintf("incidence by period of %d days",
                         x$interval)
       }
     } else if (is.character(x$interval)) {
-      x$interval <- gsub("^([a-z]+?)s*$", "\\1", x$interval)
-      ylab <- switch(x$interval,
```
Comment (caijun) [inline R/plot.R]: Here, the `ylab` string can be used for showing the bin sizes in the print info.

**Comment by zkamvar** [comment]
> For the print info for incidence object, I propose that the actual date range of observational cases is also printed. Currently only the date range used for computing the bin size is printed. This info will remind the user the difference between the two date ranges.
> 
> Taking `print(iso <- incidence(dat, "isoweek"))` as an illustration, I expect the following print info. I believe this print info is much clear.
> 
> ```
> #> <incidence object>
> #> [5888 cases observed from days 2014-04-07 to 2015-04-30]
> #> [weekly bins (or bin size in days) computed from days 2014-04-07 to 2015-04-27]
> #> [weekly bins computed from ISO weeks 2014-W15 to 2015-W18]
> ```

That's not a bad idea, but this is currently not possible since the dates are binned before they are placed in the object. This also happens if the user uses any binning higher than 1 day. To achieve the suggested behaviour, we would need to add the original date range as another item or attribute in the incidence object. I would say that this is a bit out of scope of the PR, but would be a good suggestion to add to the issue tracker wishlist

File: R/incidence.R [hunk 2]
```diff
@@ -62,18 +70,53 @@
...
+#'  - interval = "1 F week" (Friday 2016-09-02)
+#'  - interval = "1 Saturday week" (Saturday 2016-09-03)
+#'  - interval = "Sunday week" (Sunday 2016-09-04)
+#'
+#' It's also possible to use something like "3 weeks: Saturday"; In addition,
+#' there are keywords reserved for specific days of the week:
+#'   
+#'   - interval = "week", standard = TRUE (Default, Monday)
+#'   - interval = "ISOweek"  (Monday)
+#'   - interval = "EPIweek"  (Sunday)
```
[the remark this answers is not in the document]
Comment (zkamvar) [reply R/incidence.R]: I was thinking about this, but I like keeping EPIweek as an option for the following reasons:

 - epiweek is more familiar to field epis than MMWRweek
 - epiweek is easier to remember
 - the overhead for including epiweek as an option is literally a single line of code

[the remark this answers is not in the document]
Comment (zkamvar) [reply R/incidence.R]: I was thinking about this, but I like keeping EPIweek as an option for the following reasons:

 - epiweek is more familiar to field epis than MMWRweek
 - epiweek is easier to remember
 - the overhead for including epiweek as an option is literally a single line of code

File: R/plot.R (hunk 1 above)
Comment (zkamvar) [reply R/plot.R]: I'm not sure I follow. I've updated the ylab to take into account intervals like "2 months".

File: R/incidence.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (aspina7) [reply R/incidence.R]: Not convinced its worth having named intervals, why not just have start day and specify how long you want it to be? e.g. monday and 7 days...

**Comment by zkamvar** [comment]
> Not convinced its worth having named intervals, why not just have start day and specify how long you want it to be? e.g. monday and 7 days...
> 
The users can already specify something like “2 Monday weeks” if they want to. Implementing the epi, mmwr, and iso keywords is an extra. 

The thing is, it’s really not much overhead for this implementation and it aligns with existing methods in the lubridate, ISOweek, and EPIweek packages.

**Pull request merged**